app/views/root/_gem_base.html.erb

```diff
@@ -32,6 +32,14 @@
   global_bar << global_banner if global_banner
 %>
 
+<% if ENV["GOOGLE_TAG_MANAGER_ID"] %>
+   <%= render "govuk_publishing_components/components/google_tag_manager_script", {
+     gtm_id: ENV["GOOGLE_TAG_MANAGER_ID"],
+     gtm_auth: ENV["GOOGLE_TAG_MANAGER_AUTH"],
+     gtm_preview: ENV["GOOGLE_TAG_MANAGER_PREVIEW"]
+   } %>
+ <% end %>
```

Comment:
Putting the component here means that it's placed before the opening `html` tag:
<img width="1166" alt="" src="https://user-images.githubusercontent.com/1732331/155114139-d54d3693-29ca-471c-b3e2-b539e7715c72.png">

For context, we're aiming to reduce our reliance on `static` for templates - the plan is for the templates to be used directly by a frontend application rather than from `static`. 

With this in mind, I'd suggest that this should be added as an option for the public layout component in the components gem, and either the environment variables used in the component if they can be access there or passed into the component parameters. 

Happy to talk this through if there are other pressures that I'm not aware of.